cmds/record.c

```diff
@@ -1497,6 +1494,52 @@ static void print_child_usage(struct rusage *ru)
 # define EM_AARCH64  183
 #endif
 
+static bool is_regular_executable(const char *pathname)
+{
+	struct stat sb;
+
+	if (!stat(pathname, &sb))
+		if (S_ISREG(sb.st_mode) && (sb.st_mode & S_IXUSR))
+			return true;
+	return false;
+}
+
+static void find_in_path(char **exename)
+{
+	/* try to find the binary in PATH */
+	char *pathname = NULL;
+	char *env = getenv("PATH");
+	char *fullpath;
+	char *path;
+	char *p;
+	bool found = false;
+
+	if (!env || *exename[0] == '/')
+		pr_err_ns(UFTRACE_MSG, *exename);
+
+	path = fullpath = xstrdup(env);
+
+	/* search opts->exename in PATH one by one */
+	p = strchr(path, ':');
+	while (p) {
```

Comment:
You'd better using `strv` for this kind of string handling.

Reply:
Applied.